src/components/molecules/Monaco/Monaco.tsx

```diff
@@ -177,7 +177,7 @@ const Monaco = (props: {editorHeight: string; diffSelectedResource: () => void;
     setDirty(false);
 
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [selectedPath, selectedResourceId]);
+  }, [selectedPath, selectedResourceId ? resourceMap[selectedResourceId] : undefined]);
```

Comment:
How does this work?
Could you instead do:
```const selectedResource = useMemo(() => selectedResourceId ? resourceMap[selectedResourceId] : undefined), [selectedResourceId, resourceMap]);``` ?

Reply:
Better yet, create another useEffect that listens only to the `selectedResource` and just updates the code? the other things should be done only when `selectedResourceId` changes.